### PR TITLE
chore: enhance object storage error message when creating meta node

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
close https://github.com/risingwavelabs/risingwave/issues/10996

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/hummock/manager/mod.rs [hunk 1]
```diff
@@ -2748,7 +2757,7 @@ async fn write_exclusive_cluster_id(
     } else {
         let cluster_id = object_store.read(&cluster_id_full_path, None).await?;
         Err(ObjectError::internal(format!(
-            "data directory is already used by another cluster with id {:?}",
+            "Data directory is already used by another cluster with id {:?}, please try again after deleting the `data_directory/cluster_id/0.data`.",
```
Comment (Gun9niR) [inline src/meta/src/hummock/manager/mod.rs:2760]: Using the constants `CLUSTER_ID_DIR` and `CLUSTER_ID_NAME` in a format string can be more robust.

File: src/meta/src/hummock/manager/mod.rs [hunk 2]
```diff
@@ -2738,7 +2738,16 @@ async fn write_exclusive_cluster_id(
...
-    let metadata = object_store.list(&cluster_id_dir).await?;
+    let metadata = match object_store.list(&cluster_id_dir).await {
+        Ok(metadata) => metadata,
+        Err(_) => {
+            return Err(ObjectError::internal(
+                "Fail to access remote object storage, 
+            please check if your Access Key and Secret Key are configured correctly. ",
+            )
+            .into())
+        }
```
Comment (Gun9niR) [inline src/meta/src/hummock/manager/mod.rs:2749]: This failure can happen for other reasons. Shall we just match S3's `DispatchError`?

**Review by Gun9niR — APPROVED** [review]
Rest LGTM

File: src/meta/src/hummock/manager/mod.rs (hunk 2 above)
Comment (wcy-fdu) [reply src/meta/src/hummock/manager/mod.rs:2749]: The high probability of the error occurs here is that the configuration is incorrect, I just want to remind the user to check ak/sk, and make it simple. And the object storage is not only s3.

**Pull request merged**